### ISSUE TITLE
Add new functionalities to list of supported functionalities

### ISSUE THREAD

**Issue by arthurvd** [Deltares/HYDROLIB-core#466]
**What is the need for this task.**
Our documentation website contains a list of supported functionalities, on:
https://deltares.github.io/HYDROLIB-core/latest/topics/dhydro_support/

The first upcoming release will contain some new functionalities, so this list must be made consistent with these.

**What is the task?**
**Task1:**
Some recently added new functionalities should be added to the list, specifically (but only after checking that the underlying issues have been closed as Done):
* tim files (#348), should be **added** to the list, underneath `* .bc file` and the items that are contained in the `.bc` file.
* Observation crosssection file (old) (#364 + #461)
* Observation station file (old) (#459)
* External forcings file (old) (#369)

**Task 2:**
The apidocs links are wrong since the latest package restructuring.
For example, Bridge now has a dead link to `[Bridge][hydrolib.core.io.structure.models.Bridge]`, this should be `[Bridge][hydrolib.core.dflowfm.structure.Bridge]`. This probably holds for all class links.

**Additional context**
The list is maintained via `docs\topics\dhydro_support_hydrolib-core.xlsx`, only edit the "Source table" tab.

**Comment by tim-vd-aardweg** [comment]
Done:
- Refactored the sheet so that it should no longer miss any data from the master sheet.
- Updated the ExtModel section and subsections.
- Added an entry for `Observation crosssection file (.pli)`.
- Added an entry for `Observation station file (.xyn)`.
- Added a placeholder entry for the `.tim` files.
- Added a placeholder entry for the `External forcings file (old)`.

To do:
- Once #348 has been merged, we have to verify/use the correct namespace in the sheet.
- Once #369 has been merged, we have to verify/use the correct namespace in the sheet.

**Comment by MRVermeulenDeltares** [comment]
walked through the list after mkdocs serve with the branch and notice the following:

DIMR
- dimr_config.xml under functionality is differntly shown compared to other components

FM
- Network file _net.nc under functionality is differntly shown compared to other components
- Structure file under functionality is bold
- StructureModel linked to Structure file is this correct?
- is the "* indicates a scheduled release that is not out yet." still valid for structures?
- External forcings file empty, is this correct?
- External forcings file (old) missing, as expected since it is not yet in main.
- Meteo missing, as expected since it is not yet in main.
- Cross section files only got supported shown.
- Spatial data files empty
- output empty
- Observation station file (.xyn) missing, as expected since it is not yet in main.
- Observation crosssection file (.ini) missing.
- History file _his.nc under functionality is differntly shown compared to other components
- Map file (UGRID) _map.nc under functionality is differntly shown compared to other components
- Fourier output file _fou.nc under functionality is differntly shown compared to other components

RR
- Topology layer empty

link for others in list were working

**Comment by MRVermeulenDeltares** [comment]
Observation crosssection file (.ini) seems to be missing because the location is not referenced in any md file.

hydrolib.core.dflowfm.obscrosssection.models.ObservationCrossSectionModel is expected to be refernced in a md file as followed:

`::: hydrolib.core.dflowfm.obscrosssection.models`

**Comment by tim-vd-aardweg** [comment]
Created new issue for the points that Marlon brought up: https://github.com/Deltares/HYDROLIB-core/issues/518

**Comment by arthurvd** [comment]
I solved the missing obs refs that @MRVermeulenDeltares discovered:
> Observation crosssection file (.ini) seems to be missing because the location is not referenced in any md file.  
> (#364   + #461  )
_Originally posted by @MRVermeulenDeltares in https://github.com/Deltares/HYDROLIB-core/issues/466#issuecomment-1500261039_
            
It's commit https://github.com/Deltares/HYDROLIB-core/pull/479/commits/82867e95e85ad223988585e572b68f0f237b3666
